### DISCUSSION TITLE
Plano de estudos.

### DISCUSSION THREAD

**Discussion by ghost** [frontendbr/forum#725] [Discussão]
Bem, depois de algumas dicas aqui no fórum sobre como vencer a procrastinação  estou montando um plano de estudos da seguinte maneira.  Por exemplo:  

_**Segunda feira : HTML5  & CSS3
Terça-Feira: JavaScript 
Quarta-Feira: Node.Js**_ 

Seria muita coisa para focar? Ou está dentro?  Estou pensando nisso pensando de uma maneira em que: Juntando os conceitos básicos aprendidos eu consiga  criar pequenos projetos baseado nisso. O que vocês acham? Obrigado desde já.

**Comment by felipejsm** [comment]
Se é muita coisa? Talvez sim, talvez não.
Creio que dependa de quanto tempo você vai dedicar a cada conceito/conteúdo, de como você vai progredir e, principalmente, qual seu objetivo.
A ideia de entender os conceitos e aplicá-los em pequenos projetos é muito boa, ao meu ver.
Acho que o caminho é por aí mesmo, conforme você for avançando você sente o "feeling"
de ser muito ou pouco.
Como adendo, segue um texto muito bom do Guilherme Silveira sobre estudo de tecnologias: [http://blog.alura.com.br/o-dia-a-dia-de-um-profissional-que-aprende-o-calendario-de-estudo/](http://blog.alura.com.br/o-dia-a-dia-de-um-profissional-que-aprende-o-calendario-de-estudo/).

Bons estudos pra ti!

**Comment by ghost** [comment]
Eu fiz algo pensando  nesse texto mesmo rs 
Estava pensando em colocar o **MongoDB** no meio disso mas como não tenho conhecimento nenhum de banco de dados não sei se vale a pena. Pois estou seguindo algo que é usado e não está nada muito fora do Front End.   @felipejsm

**Comment by nurycaroline** [comment]
O que me ajudou foi seguir um 'tema' até terminar. 1 semana só estudando JS , na próxima Node, e assim em diante. E no meio disso tento intercalar com desafios para treinar. Tipo o Javascript 30, Daily React, e no Free code camp, tem exercícios legais também.

**Comment by AmandaFonseca** [comment]
Eu tenho esse problema com a procrastinação também.
Mas eu não consigo aprender uma nova linguagem de programação ou framework se eu não colocar a mão na massa ou seja, um livro de linguagem de  programação  não me ajuda em quase nada, só estudo se tiver com uma máquina por perto. Agora os estudos mais teóricos tipo de matemática, português, lógica de programação, engenharia de software eu  consigo. Tem um aplicativo na play store chama AnkiDroid https://play.google.com/store/apps/details?id=com.ichi2.anki&hl=pt que é excelente ele faz um pergunta e se você errar ele refaz a pergunta depois de um tempo até você fixar o conteúdo.

**Comment by simaodeveloper** [comment]
@brunojokubauskas faz o seguinte, isso irá te ajudar, segue esses dois artigos e seja feliz:

- [do-zero-a-heroi-do-front-end-parte-1](https://www.felipefialho.com/blog/2016/do-zero-a-heroi-do-front-end-parte-1)
- [do-zero-a-heroi-do-front-end-parte-2](https://www.felipefialho.com/blog/2016/do-zero-a-heroi-do-front-end-parte-2)

**Comment by ghost** [comment]
Cara, tenho grandes problemas com livros também, tanto que eu assino serviço do Alura.  Porém,  as vezes tenho a sensação que não é algo muito "completo" e só o básico para uma noção boa e logo, livros seriam necessários.  Por exemplo, algo  que fico na dúvida  é, **como vou saber a margin utilizada?** Isso é algo que me deixa na duvida em vários quesitos de CSS e fico bem perdido em tentar começar um projeto.  E acho que muitas vezes um livro seria o ideal. Mas até hoje não tenho saco de pegar algo para ler, por mais necessário que seja. Creio que um conteúdo de um livro é mais completo que um curso em vídeo.  Mas a procrastinação de ler algo, é grande demais, quero perder e criar esse hábito. 
Valeu pelo artigo @simaodeveloper  vou ler ambos! (:

**Comment by simaodeveloper** [comment]
@brunojokubauskas o que tenho a lhe dizer é "VÁ COM CALMA", você deve estar com uma sensação de que precisa absorver todo o conhecimento do mundo que esta esquecendo de fazer o básico e subir degrau por degrau, o que vejo nas suas palavras são desculpas que um dia eu já dei, você consegue sim, mas tem que firmar seu objetivo, faça uma coisa de cada vez e verá que logo o objetivo se aproxima e ai você adiciona outra meta e vai adquirindo conhecimento e experiência aos poucos, sugiro procurar o @woliveiras para trocar uma idéia, ele esta por trás do trainning-center, pelo o que vejo tu precisa de um mentor para ajudar a se localizar, participe de meetups tambem faça amizade e depois peça ajuda para te guiar via rede-social mesmo.

**Comment by ghost** [comment]
Cara, acho que você pegou bem pelo que estou passando. Estou fazendo faculdade de ADS porém já quero estudar algo por fora.  E fico muito perdido, vivo pesquisando, com várias duvidas. Eu sofro de ansiedade e tenho tido muitas crises, pois fico ansioso para aprender, mas fico muito empacado em como sair dessa. Creio que isso é normal. Estou assim até com as matérias da faculdade.   Valeu pelas dicas @simaodeveloper  vou conversar sim com o @woliveiras (:

**Comment by alinebastos** [comment]
@brunojokubauskas eu passo pelas mesmas coisas, também tenho tido crises de ansiedade e acabo não conseguindo ir adiante nos estudos ou fico sem saber pra onde ir. Te entendo perfeitamente!
O melhor conselho que eu posso te dar, que o @simaodeveloper te deu, é falar com o @woliveiras ou já entrar direto no slack do Training Center, se tu já não entrou. Lá tem um pessoal ótimo, todos se ajudam, tem muita gente começando também! Conta com a gente!
https://ctgroups.herokuapp.com/

**Comment by woliveiras** [comment]
@brunojokubauskas, seu plano de estudos me parece bom, mas está muito vago (talvez você resumiu pra caber aqui).

Uma dica é: monte tudo em tópicos pequenos e faça projetos a cada final de conteúdo.

Ex.: **Aprender HTML**

Você vai precisar conhecer algumas tags que sempre são utilizadas pra criar um projeto, certo?
Então seu trabalho inicial deve ser pesquisar o que você precisa aprender de HTML e separar em sub-tópicos. Para cada sub-tópico você pode colocar alguns minutos pra estudar e quando finalizar tudo o que precisa aprender de HTML, então cria um projeto simples (sem CSS mesmo, você pode validar seu HTML com ferramentas de SEO e Acessibilidade) e assim não estudar muita coisa de uma vez e só ficar perdido por ter muito conteúdo teórico na cabeça e pouco prático.

Depois que finalizar esses tópicos, então você monta o tópico do CSS e seus sub-tópicos, em seguida o do JavaScript, do Node e de **bancos de dados** (não aprenda somente MongoDB). Algo importante que você poderia adicionar em sua trilha depois que finalizar tudo são testes unitários e TDD.

Como você está fazendo um plano seguindo datas certinhas pra estudar, eu recomendo que pegue uma leva de sub-tópicos e estude no dia do HTML, depois faça o mesmo pra JavaScript, etc.

Mas cuidado com uma coisa: o descanso é algo muito importante no aprendizado. Você precisa tirar dias para descansar e descansar não é ficar em casa vendo série, se você faz isso todo final de semana, tem que sair da rotina. Quanto mais disposto você estiver, mais coragem vai ter pra encarar os estudos.

Os links que o @simaodeveloper mandou são excelentes, eu sempre indico pra todo mundo, e a dica da @AmandaFonseca de usar o Anki é super boa!

Você pode anotar todos esses sub-tópicos no Anki e ir forçando a recuperação do conteúdo na memória até gravar na memória de longo prazo.

Eu também sempre indico o [freeCodeCamp](https://medium.com/trainingcenter/aprenda-programa%C3%A7%C3%A3o-do-iniciante-ao-avan%C3%A7ado-com-um-curso-focado-no-mercado-de-trabalho-858c4b0928ed), pois no final de cada módulo tem um exercício prático e sempre te forçando a criar o portfólio.

Se manter motivado a estudar é o principal. Você precisa ter uma "recompensa" no final da trilha e por isso é importante que você não fique só nos estudos, vá a eventos conhecer pessoas e se não der pra fazer isso participe das coisas online. Nosso [Slack](https://github.com/training-center/slack) é excelente pra encontrar motivação, sempre tem alguém lá pra te botar pra cima.

Chega junto da gente lá! :smile:

**Comment by didiraja** [comment]
Opa, Bruno. Estou nessa fase de estudo também, atualmente saindo do JS e descobrindo usos de jQuery e Ajax. O que mais funcionou pra mim foram essas metodologias:

**1. Focar em um módulo de estudo por vez** (Não faço essa divisão manualmente, os curso da Alura me ajudam com isso). Ajuda a não confundir os conhecimento. Sendo que no sue caso especificamente, Javascript é um conteúdo gigante por is e pra usar Node.js você precisa de um conhecimento no mínimo mediano pra saber mexer.

2. **Assistir por completo o vídeo explicativo e só depois fazer o exercício**, agora sim, acompanhado do vídeo (basta ler o material escrito, caso tenha)

3. **Não se cobrar muito em fazer a aula/etapa toda de uma vez no dia**. Acomode ela ao seu cotidiano. Pra mim funcionou muito melhor ver o vídeo no onibus e depois fazer o exercicio quando podia (seja mais tarde naquele dia, ou em outro dia mesmo)  do que esperar eu ter 1-2 horas disponíveis pra fazer tudo de uma vez (vida agitada, hehe)

**Comment by JeanLLopes** [comment]
Ola @brunojokubauskas, muito bacana essa atitude de montar esse plano de estudos, vendo seu plano posso de dar uma dica, tente colocar um dos dias para se dedicar ao ingles, pois muitos artigos bons sobre o que vc pretende aprender tambem podem estar em ingles..
Claro vale essa dica caso nao tenha dominio do ingles..

**Comment by victorsilent** [comment]
Eu utilizava essa metodologia de estudar uma coisa por dia e bem... Deu certo por 3 meses (os três meses mais produtivos da minha vida) mas depois disso, juntando a rotina da faculdade e etc, ficou insustentavel, dai parti pra ver uma coisa por vez, é mais fácil você não cai em problemas como "parei onde ontem?", "Isso foi em Node ou JS mesmo?", "Isso não é uma parada do CSS?" e, pelo menos pra mim, ficou mais simples de solidificar um conhecimento.

**Comment by yanmagale** [comment]
O método é bem legal ... mas assim, evidentemente que cada um é cada um. Meu conselho seria para você testar diversos métodos e ver com qual deles você se adapta melhor.


Mas sem dúvida alguma, ter um schedule é bem importante. A medida que você reserva tempo para fazer aquela tarefa, a constância desta ação irá criar o hábito. E isto é o mais difícil.

Muita força e conta com a gente :smile:
